// ionic-angular/components/navbar/navbar.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/render/api';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/navbar/navbar';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/components/app/app';
import * as import9 from 'ionic-angular/navigation/view-controller';
import * as import10 from 'ionic-angular/navigation/nav-controller';
import * as import11 from 'ionic-angular/config/config';
import * as import12 from '@angular/core/src/linker/element_ref';
import * as import13 from '@angular/core/src/metadata/view';
import * as import14 from '@angular/core/src/linker/component_factory';
import * as import15 from '@angular/core/src/linker/query_list';
import * as import16 from '@angular/common/src/directives/ng_class';
import * as import17 from 'ionic-angular/components/button/button';
import * as import18 from 'ionic-angular/components/icon/icon';
import * as import19 from '@angular/core/src/change_detection/differs/iterable_differs';
import * as import20 from '@angular/core/src/change_detection/differs/keyvalue_differs';
import * as import21 from '../button/button.ngfactory';
var renderType_Navbar_Host:import0.RenderComponentType = (null as any);
class _View_Navbar_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import2.AppElement;
  _Navbar_0_4:import3.Navbar;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_Navbar_Host0,renderType_Navbar_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.selectOrCreateHostElement('ion-navbar',rootSelector,(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','toolbar');
    this._appEl_0 = new import2.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_Navbar0(this.viewUtils,this.injector(0),this._appEl_0);
    this._Navbar_0_4 = new import3.Navbar(this.parentInjector.get(import8.App),this.parentInjector.get(import9.ViewController,(null as any)),this.parentInjector.get(import10.NavController,(null as any)),this.parentInjector.get(import11.Config),new import12.ElementRef(this._el_0),this.renderer);
    this._appEl_0.initComponent(this._Navbar_0_4,([] as any[]),compView_0);
    compView_0.create(this._Navbar_0_4,this.projectableNodes,(null as any));
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],([] as any[]),([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.Navbar) && (0 === requestNodeIndex))) { return this._Navbar_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = this._Navbar_0_4._hidden;
    if (import4.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementProperty(this._el_0,'hidden',currVal_0);
      this._expr_0 = currVal_0;
    }
    const currVal_1:any = this._Navbar_0_4._sbPadding;
    if (import4.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementClass(this._el_0,'statusbar-padding',currVal_1);
      this._expr_1 = currVal_1;
    }
    this.detectViewChildrenChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Navbar_0_4.ngAfterViewInit(); } }
  }
}
function viewFactory_Navbar_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  if ((renderType_Navbar_Host === (null as any))) { (renderType_Navbar_Host = viewUtils.createRenderComponentType('',0,import13.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_Navbar_Host0(viewUtils,parentInjector,declarationEl);
}
export const NavbarNgFactory:import14.ComponentFactory<import3.Navbar> = new import14.ComponentFactory<import3.Navbar>('ion-navbar',viewFactory_Navbar_Host0,import3.Navbar);
const styles_Navbar:any[] = ([] as any[]);
var renderType_Navbar:import0.RenderComponentType = (null as any);
class _View_Navbar0 extends import1.AppView<import3.Navbar> {
  _viewQuery_bbTxt_0:import15.QueryList<any>;
  _el_0:any;
  _NgClass_0_3:import16.NgClass;
  _el_1:any;
  /*private*/ _appEl_1:import2.AppElement;
  _NgClass_1_4:import16.NgClass;
  _Button_1_5:import17.Button;
  _el_2:any;
  _el_3:any;
  _NgClass_3_3:import16.NgClass;
  _Icon_3_4:import18.Icon;
  _el_4:any;
  _NgClass_4_3:import16.NgClass;
  _el_5:any;
  _NgClass_5_3:import16.NgClass;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_6:any;
  /*private*/ _expr_7:any;
  /*private*/ _expr_8:any;
  /*private*/ _expr_9:any;
  /*private*/ _expr_10:any;
  /*private*/ _expr_11:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_Navbar0,renderType_Navbar,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._viewQuery_bbTxt_0 = new import15.QueryList<any>();
    this._el_0 = this.renderer.createElement(parentRenderNode,'div',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','toolbar-background');
    this._NgClass_0_3 = new import16.NgClass(this.parentInjector.get(import19.IterableDiffers),this.parentInjector.get(import20.KeyValueDiffers),new import12.ElementRef(this._el_0),this.renderer);
    this._el_1 = this.renderer.createElement(parentRenderNode,'button',(null as any));
    this.renderer.setElementAttribute(this._el_1,'class','back-button');
    this.renderer.setElementAttribute(this._el_1,'ion-button','bar-button');
    this._appEl_1 = new import2.AppElement(1,(null as any),this,this._el_1);
    var compView_1:any = import21.viewFactory_Button0(this.viewUtils,this.injector(1),this._appEl_1);
    this._NgClass_1_4 = new import16.NgClass(this.parentInjector.get(import19.IterableDiffers),this.parentInjector.get(import20.KeyValueDiffers),new import12.ElementRef(this._el_1),this.renderer);
    this._Button_1_5 = new import17.Button((null as any),'bar-button',this.parentInjector.get(import11.Config),new import12.ElementRef(this._el_1),this.renderer);
    this._appEl_1.initComponent(this._Button_1_5,([] as any[]),compView_1);
    this._el_2 = this.renderer.createElement((null as any),'span',(null as any));
    this.renderer.setElementAttribute(this._el_2,'class','button-inner');
    this._el_3 = this.renderer.createElement(this._el_2,'ion-icon',(null as any));
    this.renderer.setElementAttribute(this._el_3,'class','back-button-icon');
    this.renderer.setElementAttribute(this._el_3,'role','img');
    this._NgClass_3_3 = new import16.NgClass(this.parentInjector.get(import19.IterableDiffers),this.parentInjector.get(import20.KeyValueDiffers),new import12.ElementRef(this._el_3),this.renderer);
    this._Icon_3_4 = new import18.Icon(this.parentInjector.get(import11.Config),new import12.ElementRef(this._el_3),this.renderer);
    this._el_4 = this.renderer.createElement(this._el_2,'span',(null as any));
    this.renderer.setElementAttribute(this._el_4,'class','back-button-text');
    this._NgClass_4_3 = new import16.NgClass(this.parentInjector.get(import19.IterableDiffers),this.parentInjector.get(import20.KeyValueDiffers),new import12.ElementRef(this._el_4),this.renderer);
    compView_1.create(this._Button_1_5,[([] as any[]).concat([this._el_2])],(null as any));
    this.renderer.projectNodes(parentRenderNode,import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
    this.renderer.projectNodes(parentRenderNode,import4.flattenNestedViewRenderNodes(this.projectableNodes[1]));
    this.renderer.projectNodes(parentRenderNode,import4.flattenNestedViewRenderNodes(this.projectableNodes[2]));
    this._el_5 = this.renderer.createElement(parentRenderNode,'div',(null as any));
    this.renderer.setElementAttribute(this._el_5,'class','toolbar-content');
    this._NgClass_5_3 = new import16.NgClass(this.parentInjector.get(import19.IterableDiffers),this.parentInjector.get(import20.KeyValueDiffers),new import12.ElementRef(this._el_5),this.renderer);
    this.renderer.projectNodes(this._el_5,import4.flattenNestedViewRenderNodes(this.projectableNodes[3]));
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    var disposable_0:Function = this.renderer.listen(this._el_1,'click',this.eventHandler(this._handle_click_1_0.bind(this)));
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_6 = import7.UNINITIALIZED;
    this._expr_7 = import7.UNINITIALIZED;
    this._expr_8 = import7.UNINITIALIZED;
    this._expr_9 = import7.UNINITIALIZED;
    this._expr_10 = import7.UNINITIALIZED;
    this._expr_11 = import7.UNINITIALIZED;
    this._expr_12 = import7.UNINITIALIZED;
    this._expr_13 = import7.UNINITIALIZED;
    this._viewQuery_bbTxt_0.reset([new import12.ElementRef(this._el_4)]);
    this.context._bbTxt = this._viewQuery_bbTxt_0.first;
    this.init(([] as any[]),[
      this._el_0,
      this._el_1,
      this._el_2,
      this._el_3,
      this._el_4,
      this._el_5
    ]
    ,[disposable_0],([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import16.NgClass) && (0 === requestNodeIndex))) { return this._NgClass_0_3; }
    if (((token === import16.NgClass) && (3 === requestNodeIndex))) { return this._NgClass_3_3; }
    if (((token === import18.Icon) && (3 === requestNodeIndex))) { return this._Icon_3_4; }
    if (((token === import16.NgClass) && (4 === requestNodeIndex))) { return this._NgClass_4_3; }
    if (((token === import16.NgClass) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._NgClass_1_4; }
    if (((token === import17.Button) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 4)))) { return this._Button_1_5; }
    if (((token === import16.NgClass) && (5 === requestNodeIndex))) { return this._NgClass_5_3; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = 'toolbar-background';
    if (import4.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this._NgClass_0_3.klass = currVal_0;
      this._expr_0 = currVal_0;
    }
    const currVal_1:any = ('toolbar-background-' + this.context._mode);
    if (import4.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this._NgClass_0_3.ngClass = currVal_1;
      this._expr_1 = currVal_1;
    }
    if (!throwOnChange) { this._NgClass_0_3.ngDoCheck(); }
    const currVal_4:any = 'back-button';
    if (import4.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this._NgClass_1_4.klass = currVal_4;
      this._expr_4 = currVal_4;
    }
    const currVal_5:any = ('back-button-' + this.context._mode);
    if (import4.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this._NgClass_1_4.ngClass = currVal_5;
      this._expr_5 = currVal_5;
    }
    if (!throwOnChange) { this._NgClass_1_4.ngDoCheck(); }
    const currVal_6:any = 'back-button-icon';
    if (import4.checkBinding(throwOnChange,this._expr_6,currVal_6)) {
      this._NgClass_3_3.klass = currVal_6;
      this._expr_6 = currVal_6;
    }
    const currVal_7:any = ('back-button-icon-' + this.context._mode);
    if (import4.checkBinding(throwOnChange,this._expr_7,currVal_7)) {
      this._NgClass_3_3.ngClass = currVal_7;
      this._expr_7 = currVal_7;
    }
    if (!throwOnChange) { this._NgClass_3_3.ngDoCheck(); }
    const currVal_8:any = this.context._bbIcon;
    if (import4.checkBinding(throwOnChange,this._expr_8,currVal_8)) {
      this._Icon_3_4.name = currVal_8;
      this._expr_8 = currVal_8;
    }
    const currVal_10:any = 'back-button-text';
    if (import4.checkBinding(throwOnChange,this._expr_10,currVal_10)) {
      this._NgClass_4_3.klass = currVal_10;
      this._expr_10 = currVal_10;
    }
    const currVal_11:any = ('back-button-text-' + this.context._mode);
    if (import4.checkBinding(throwOnChange,this._expr_11,currVal_11)) {
      this._NgClass_4_3.ngClass = currVal_11;
      this._expr_11 = currVal_11;
    }
    if (!throwOnChange) { this._NgClass_4_3.ngDoCheck(); }
    const currVal_12:any = 'toolbar-content';
    if (import4.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this._NgClass_5_3.klass = currVal_12;
      this._expr_12 = currVal_12;
    }
    const currVal_13:any = ('toolbar-content-' + this.context._mode);
    if (import4.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this._NgClass_5_3.ngClass = currVal_13;
      this._expr_13 = currVal_13;
    }
    if (!throwOnChange) { this._NgClass_5_3.ngDoCheck(); }
    this.detectContentChildrenChanges(throwOnChange);
    if (!throwOnChange) { if ((this.numberOfChecks === 0)) { this._Button_1_5.ngAfterContentInit(); } }
    const currVal_3:any = this.context._hideBb;
    if (import4.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementProperty(this._el_1,'hidden',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_9:any = this._Icon_3_4._hidden;
    if (import4.checkBinding(throwOnChange,this._expr_9,currVal_9)) {
      this.renderer.setElementClass(this._el_3,'hide',currVal_9);
      this._expr_9 = currVal_9;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._Icon_3_4.ngOnDestroy();
  }
  private _handle_click_1_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.backButtonClick($event)) !== false);
    return (true && pd_0);
  }
}
export function viewFactory_Navbar0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<import3.Navbar> {
  if ((renderType_Navbar === (null as any))) { (renderType_Navbar = viewUtils.createRenderComponentType('',4,import13.ViewEncapsulation.None,styles_Navbar,{})); }
  return new _View_Navbar0(viewUtils,parentInjector,declarationEl);
}
