// @ionic/app-scripts/dist/util/events.js
"use strict";
var events_1 = require('events');
var emmitter = new events_1.EventEmitter();
function on(event, listener) {
    return emmitter.on('event' + event.toString(), listener);
}
exports.on = on;
function emit(event, context, arg) {
    return emmitter.emit('event' + event.toString(), context, arg);
}
exports.emit = emit;
(function (EventType) {
    EventType[EventType["FileChange"] = 0] = "FileChange";
    EventType[EventType["FileAdd"] = 1] = "FileAdd";
    EventType[EventType["FileDelete"] = 2] = "FileDelete";
    EventType[EventType["DirectoryAdd"] = 3] = "DirectoryAdd";
    EventType[EventType["DirectoryDelete"] = 4] = "DirectoryDelete";
    EventType[EventType["BundlerStart"] = 5] = "BundlerStart";
    EventType[EventType["BunderFinish"] = 6] = "BunderFinish";
    EventType[EventType["TranspilerStart"] = 7] = "TranspilerStart";
    EventType[EventType["TranspilerFinish"] = 8] = "TranspilerFinish";
    EventType[EventType["SassStart"] = 9] = "SassStart";
    EventType[EventType["SassFinish"] = 10] = "SassFinish";
})(exports.EventType || (exports.EventType = {}));
var EventType = exports.EventType;
;
