// ionic-angular/components/item/item-reorder.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/item/item-reorder';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/components/item/item';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
import * as import12 from 'ionic-angular/components/icon/icon';
import * as import13 from 'ionic-angular/config/config';
var renderType_Reorder_Host = null;
var _View_Reorder_Host0 = (function (_super) {
    __extends(_View_Reorder_Host0, _super);
    function _View_Reorder_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Reorder_Host0, renderType_Reorder_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Reorder_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-reorder', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Reorder0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Reorder_0_4 = new import3.Reorder(this.parentInjector.get(import8.Item), new import9.ElementRef(this._el_0));
        this._appEl_0.initComponent(this._Reorder_0_4, [], compView_0);
        compView_0.create(this._Reorder_0_4, this.projectableNodes, null);
        var disposable_0 = this.renderer.listen(this._el_0, 'click', this.eventHandler(this._handle_click_0_0.bind(this)));
        this.init([].concat([this._el_0]), [this._el_0], [disposable_0], []);
        return this._appEl_0;
    };
    _View_Reorder_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Reorder) && (0 === requestNodeIndex))) {
            return this._Reorder_0_4;
        }
        return notFoundResult;
    };
    _View_Reorder_Host0.prototype._handle_click_0_0 = function ($event) {
        this._appEl_0.componentView.markPathToRootAsCheckOnce();
        var pd_0 = (this._Reorder_0_4.onClick($event) !== false);
        return (true && pd_0);
    };
    return _View_Reorder_Host0;
}(import1.AppView));
function viewFactory_Reorder_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Reorder_Host === null)) {
        (renderType_Reorder_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_Reorder_Host0(viewUtils, parentInjector, declarationEl);
}
export var ReorderNgFactory = new import11.ComponentFactory('ion-reorder', viewFactory_Reorder_Host0, import3.Reorder);
var styles_Reorder = [];
var renderType_Reorder = null;
var _View_Reorder0 = (function (_super) {
    __extends(_View_Reorder0, _super);
    function _View_Reorder0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Reorder0, renderType_Reorder, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Reorder0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'ion-icon', null);
        this.renderer.setElementAttribute(this._el_0, 'name', 'reorder');
        this.renderer.setElementAttribute(this._el_0, 'role', 'img');
        this._Icon_0_3 = new import12.Icon(this.parentInjector.get(import13.Config), new import9.ElementRef(this._el_0), this.renderer);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this.init([], [this._el_0], [], []);
        return null;
    };
    _View_Reorder0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import12.Icon) && (0 === requestNodeIndex))) {
            return this._Icon_0_3;
        }
        return notFoundResult;
    };
    _View_Reorder0.prototype.detectChangesInternal = function (throwOnChange) {
        var currVal_0 = 'reorder';
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this._Icon_0_3.name = currVal_0;
            this._expr_0 = currVal_0;
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_1 = this._Icon_0_3._hidden;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_0, 'hide', currVal_1);
            this._expr_1 = currVal_1;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Reorder0.prototype.destroyInternal = function () {
        this._Icon_0_3.ngOnDestroy();
    };
    return _View_Reorder0;
}(import1.AppView));
export function viewFactory_Reorder0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Reorder === null)) {
        (renderType_Reorder = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, styles_Reorder, {}));
    }
    return new _View_Reorder0(viewUtils, parentInjector, declarationEl);
}
