// ionic-angular/components/item/item-reorder.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/render/api';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/item/item-reorder';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/components/item/item';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
import * as import12 from 'ionic-angular/components/icon/icon';
import * as import13 from 'ionic-angular/config/config';
var renderType_Reorder_Host:import0.RenderComponentType = (null as any);
class _View_Reorder_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import2.AppElement;
  _Reorder_0_4:import3.Reorder;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_Reorder_Host0,renderType_Reorder_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.selectOrCreateHostElement('ion-reorder',rootSelector,(null as any));
    this._appEl_0 = new import2.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_Reorder0(this.viewUtils,this.injector(0),this._appEl_0);
    this._Reorder_0_4 = new import3.Reorder(this.parentInjector.get(import8.Item),new import9.ElementRef(this._el_0));
    this._appEl_0.initComponent(this._Reorder_0_4,([] as any[]),compView_0);
    compView_0.create(this._Reorder_0_4,this.projectableNodes,(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_0,'click',this.eventHandler(this._handle_click_0_0.bind(this)));
    this.init(([] as any[]).concat([this._el_0]),[this._el_0],[disposable_0],([] as any[]));
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.Reorder) && (0 === requestNodeIndex))) { return this._Reorder_0_4; }
    return notFoundResult;
  }
  private _handle_click_0_0($event:any):boolean {
    this._appEl_0.componentView.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this._Reorder_0_4.onClick($event)) !== false);
    return (true && pd_0);
  }
}
function viewFactory_Reorder_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  if ((renderType_Reorder_Host === (null as any))) { (renderType_Reorder_Host = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,([] as any[]),{})); }
  return new _View_Reorder_Host0(viewUtils,parentInjector,declarationEl);
}
export const ReorderNgFactory:import11.ComponentFactory<import3.Reorder> = new import11.ComponentFactory<import3.Reorder>('ion-reorder',viewFactory_Reorder_Host0,import3.Reorder);
const styles_Reorder:any[] = ([] as any[]);
var renderType_Reorder:import0.RenderComponentType = (null as any);
class _View_Reorder0 extends import1.AppView<import3.Reorder> {
  _el_0:any;
  _Icon_0_3:import12.Icon;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_Reorder0,renderType_Reorder,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._el_0 = this.renderer.createElement(parentRenderNode,'ion-icon',(null as any));
    this.renderer.setElementAttribute(this._el_0,'name','reorder');
    this.renderer.setElementAttribute(this._el_0,'role','img');
    this._Icon_0_3 = new import12.Icon(this.parentInjector.get(import13.Config),new import9.ElementRef(this._el_0),this.renderer);
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this.init(([] as any[]),[this._el_0],([] as any[]),([] as any[]));
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import12.Icon) && (0 === requestNodeIndex))) { return this._Icon_0_3; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    const currVal_0:any = 'reorder';
    if (import4.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this._Icon_0_3.name = currVal_0;
      this._expr_0 = currVal_0;
    }
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_1:any = this._Icon_0_3._hidden;
    if (import4.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementClass(this._el_0,'hide',currVal_1);
      this._expr_1 = currVal_1;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._Icon_0_3.ngOnDestroy();
  }
}
export function viewFactory_Reorder0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<import3.Reorder> {
  if ((renderType_Reorder === (null as any))) { (renderType_Reorder = viewUtils.createRenderComponentType('',0,import10.ViewEncapsulation.None,styles_Reorder,{})); }
  return new _View_Reorder0(viewUtils,parentInjector,declarationEl);
}
