// ionic-angular/components/button/button.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/button/button';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/config/config';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/core/src/metadata/view';
import * as import11 from '@angular/core/src/linker/component_factory';
var renderType_Button_Host = null;
var _View_Button_Host0 = (function (_super) {
    __extends(_View_Button_Host0, _super);
    function _View_Button_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Button_Host0, renderType_Button_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Button_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('div', rootSelector, null);
        this.renderer.setElementAttribute(this._el_0, 'ion-button', '');
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Button0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Button_0_4 = new import3.Button(null, '', this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_0), this.renderer);
        this._appEl_0.initComponent(this._Button_0_4, [], compView_0);
        compView_0.create(this._Button_0_4, this.projectableNodes, null);
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Button_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Button) && (0 === requestNodeIndex))) {
            return this._Button_0_4;
        }
        return notFoundResult;
    };
    _View_Button_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._Button_0_4.ngAfterContentInit();
            }
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    return _View_Button_Host0;
}(import1.AppView));
function viewFactory_Button_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Button_Host === null)) {
        (renderType_Button_Host = viewUtils.createRenderComponentType('', 0, import10.ViewEncapsulation.None, [], {}));
    }
    return new _View_Button_Host0(viewUtils, parentInjector, declarationEl);
}
export var ButtonNgFactory = new import11.ComponentFactory('[ion-button]', viewFactory_Button_Host0, import3.Button);
var styles_Button = [];
var renderType_Button = null;
var _View_Button0 = (function (_super) {
    __extends(_View_Button0, _super);
    function _View_Button0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Button0, renderType_Button, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckOnce);
    }
    _View_Button0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._el_0 = this.renderer.createElement(parentRenderNode, 'span', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'button-inner');
        this.renderer.projectNodes(this._el_0, import4.flattenNestedViewRenderNodes(this.projectableNodes[0]));
        this._el_1 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'button-effect');
        this.init([], [
            this._el_0,
            this._el_1
        ], [], []);
        return null;
    };
    return _View_Button0;
}(import1.AppView));
export function viewFactory_Button0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Button === null)) {
        (renderType_Button = viewUtils.createRenderComponentType('', 1, import10.ViewEncapsulation.None, styles_Button, {}));
    }
    return new _View_Button0(viewUtils, parentInjector, declarationEl);
}
